esi_leap/resource_objects/ironic_node.py

```diff
@@ -101,6 +102,11 @@ def resource_admin_project_id(self):
         return self._get_node().owner
 
     def _get_node(self, resource_list=None):
-        if not self._node:
-            self._node = ironic.get_node(self._uuid, resource_list)
+        try:
+
+            if not self._node:
+                self._node = ironic.get_node(self._uuid, resource_list)
+        except exceptions.NotFound:
+            self._node = 'Unknown'
```

Comment:
Ah, this isn't right; self._node is an actual node object, so you can't just return a string. You'd want to set it to None, and then let the API code deal with setting the name/other attributes.

Alternatively you could create a dummy Ironic node.

Reply:
Ah, I see